catkin_virtualenv/cmake/build_venv.py

```diff
@@ -23,21 +23,36 @@
 import os
 import re
 import shutil
+import subprocess
 import sys
 
-from dh_virtualenv import Deployment
+from dh_virtualenv.deployment import Deployment
 from distutils.spawn import find_executable
 
 
 _BYTECODE_REGEX = re.compile('.*.py[co]')
 
 
 def delete_bytecode(directory):
-    for root, dirs, files in os.walk(directory):
+    for root, _, files in os.walk(directory):
         for f in files:
             if _BYTECODE_REGEX.match(f):
                 os.remove(os.path.join(root, f))
 
+def find_python(version):
+    python_executable = find_executable('python' + version)
+    if not python_executable:
+        raise RuntimeError("Unable to find python executable 'python{}''".format(args.python_version), file=sys.stderr)
+    return python_executable
+
+def check_venv_module(python_executable):
+    try:
+        with open(os.devnull, 'w') as devnull:
+            subprocess.check_call(python_executable + " -c 'import venv'", shell=True, stderr=devnull)
```

Comment:
Does this need `shell=True`? Last I checked that was unsafe.

Reply:
`shell=True` does have some issues if you don't trust your environment.

You're right though, it should work without it, however I can't figure out why it doesn't:

```python
import subprocess

cmd_arr = ['/usr/bin/python2', "-c", "'import venv'"]
cmd_str = ' '.join(cmd_arr)

subprocess.check_call(cmd_arr)  # nada
subprocess.check_call(cmd_str, shell=True)  # raises CalledProcessError as expected
```

Reply:
Looks like it's supposed to be:

```python
cmd_arr = ["/usr/bin/python2", "-c 'import venv'"]

# however, this returns a CalledProcessError for entirely the wrong reason
subprocess.check_output(cmd_arr)
  File "<string>", line 1
    'import venv'
    ^
IndentationError: unexpected indent
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/subprocess.py", line 574, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['/usr/bin/python3', "-c 'import venv'"]' returned non-zero exit status 1
```

Reply:
I'm fine for this to move forward with your current working version, though a comment explaining the weird syntax as a necessity for subprocess might be good. But I was bored, so I did a little digging for wtf is happening here:

Clearly the args must go through an extra layer of processing which doesn't handle the space properly -- i.e. it ends up splitting `-c` and ` import venv` and then tries to run ` import venv` with a one space indent. [This SO user](https://stackoverflow.com/a/805088) suggests at least as far as the Windows implementation goes:
> I believe that list2cmdline(), which is doing the processing of your list args, splits any string arg on whitespace unless the string contains double quotes.

So it's actually possible that if you changed to `'-c "import venv"'` it would work... (or used \" or something). And notice that this means the reason `shell=True` works is probably because Python isn't clumsily recreating a single string.

Reply:
`'-c "import venv"'` seems equivalent to `"-c 'import venv'"` - it does make it through subprocess okay, but reports an IntendationError.